st2actions/st2actions/query/mistral/v2.py

```diff
@@ -131,13 +131,12 @@ def _determine_execution_status(self, execution_id, wf_state, tasks):
         # Identify the list of tasks that are not in completed states.
         active_tasks = [t for t in tasks if t['state'] not in DONE_STATES]
 
-        # On cancellation, mistral workflow executions are paused so that tasks can
-        # gracefully reach completion. If any task is not completed, do not mark st2
-        # action execution for the workflow complete. By marking the st2 action execution
-        # as running, this will keep the query for this mistral workflow execution active.
-        if wf_state not in DONE_STATES and not active_tasks and is_action_canceled:
+        # On cancellation, mistral workflow executions are paused so that tasks
+        # can gracefully reach completion. This is only temporary until a canceled
+        # status is added to mistral.
+        if (wf_state in DONE_STATES or wf_state == 'PAUSED') and is_action_canceled:
```

Comment:
Hm, I'm a bit confused about this condition (maybe I'm just not reading / understanding it correctly).

What happens if workflow is in a running state and user cancels an execution? It looks like that right now we won't change the status to `canceled` in that scenario? Is this correct / desired? Or should this be `wf_state not in DONE_STATES`?